### PR TITLE
Wrap socket factory to set Endpoint Identification algo

### PR DESCRIPTION
**Pull request by mcpherrinm**

### CODE REVIEW

**Comment by mcpherrinm** [comment]
R?: @alokmenghrajani @csstaub @sqshh

File: server/src/main/java/keywhiz/auth/ldap/EndpointIdentificationSocketFactory.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * LDAP.  This is important for connecting to TLS through a load balancer, where we may see
+ * different certs, triggering Java8's 3shake mitigation */
+final class EndpointIdentificationSocketFactory extends SocketFactory {
+  private final SSLSocketFactory factory;
+
+  EndpointIdentificationSocketFactory(SSLSocketFactory factory) {
+    this.factory = factory;
+  }
+
+  private Socket setEndpoint(Socket socket) {
```
Comment (csstaub) [inline server/src/main/java/keywhiz/auth/ldap/EndpointIdentificationSocketFactory.java]: `setEndpoint` is a bit misleading name

Comment (csstaub) [inline server/src/main/java/keywhiz/auth/ldap/EndpointIdentificationSocketFactory.java]: maybe `prepareSocket`, `setParameters`, etc?

File: server/src/main/java/keywhiz/auth/ldap/EndpointIdentificationSocketFactory.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    SSLSocket sslSocket = (SSLSocket)socket;
+    SSLParameters parameters = sslSocket.getSSLParameters();
+    parameters.setEndpointIdentificationAlgorithm("LDAPS");
```
Comment (alokmenghrajani) [inline server/src/main/java/keywhiz/auth/ldap/EndpointIdentificationSocketFactory.java]: There's little difference between LDAPS and HTTPS. We could make this code more re-usable by making the algorithm a parameter?

**Comment by alokmenghrajani** [comment]
👍

File: server/src/main/java/keywhiz/auth/ldap/EndpointIdentificationSocketFactory.java (hunk 2 above)
Comment (mcpherrinm) [inline server/src/main/java/keywhiz/auth/ldap/EndpointIdentificationSocketFactory.java]: Yeah, we could -- but maybe do that when we actually need it as we'd have to make it not-private and such too.

File: server/src/main/java/keywhiz/auth/ldap/EndpointIdentificationSocketFactory.java (hunk 1 above)
Comment (mcpherrinm) [inline server/src/main/java/keywhiz/auth/ldap/EndpointIdentificationSocketFactory.java]: Yeah, that is a bad name.  I like prepareSocket; I'll rename to that.

**Pull request merged**